### PR TITLE
fix(cb2-8192): ref-data admin - adds validation for required field 'resourceKey'

### PR DESCRIPTION
**Pull request by emilyrkelly21**
There is no validation for blank resource key when adding Country of registration

[CB2-8192](https://dvsa.atlassian.net/browse/CB2-8192)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

**Review by owen-corrigan-bjss — DISMISSED** [review]

**Review by andrewf-bjss — DISMISSED** [review]

File: src/app/features/reference-data/reference-data-list/reference-data-list.component.ts [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
-  constructor(private referenceDataService: ReferenceDataService, private route: ActivatedRoute, private router: Router, private store: Store) {
-    this.referenceDataService.loadReferenceData(ReferenceDataResourceType.ReferenceDataAdminType);
+  constructor(private referenceDataService: ReferenceDataService, private route: ActivatedRoute, private router: Router, private store: Store) {}
 
-    this.route.queryParams.pipe(take(1)).subscribe(params => {
+  ngOnInit(): void {
+    this.route.params.pipe(take(1)).subscribe(params => {
       this.type = params['type'];
+
+      console.log('This is the type by url params: ', this.type);
```
Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-list/reference-data-list.component.ts:34]: Probably don't want to `console.log` here 😉

File: src/app/features/reference-data/reference-data-add/reference-data-add.component.ts [hunk 2]
```diff
@@ -39,34 +44,14 @@ export class AddReferenceDataComponent {
...
-                  type: FormNodeTypes.CONTROL,
-                  editType: (referenceData as any)[property]
-                },
-                '',
-                [Validators.required]
-              )
-            )
-          );
-      });
+      console.log('This is the type by url params: ', this.type);
```
Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-add/reference-data-add.component.ts:53]: `console.log` probably needs removing

File: src/app/features/reference-data/reference-data-add/reference-data-add.component.ts [hunk 3]
```diff
@@ -77,13 +62,52 @@ export class AddReferenceDataComponent {
...
+      default:
+        templateToReturn = {} as FormNode;
+        break;
+    }
+    templateToReturn.children?.forEach(child => {
+      if (child.name === 'resourceKey') {
+        child.disabled = false;
+      }
+    });
+    return templateToReturn;
```
Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-add/reference-data-add.component.ts:110]: I'm not sure why you need to do that if the control is disabled in the controls anyways. If you do that, you can also get rid of the `let` and the `break`s, but just returning inside of the switch statement.

Comment (gjulien-bjss) [inline src/app/features/reference-data/reference-data-add/reference-data-add.component.ts:110]: Additionally, you could return the template which has the `name` equal to the `ReferenceDataResourceType`

**Review by owen-corrigan-bjss — APPROVED** [review]

**Review by naathanbrown — APPROVED** [review]

**Pull request merged**